graylog2-web-interface/src/components/search/MessageFieldExtractorActions.jsx

```diff
@@ -25,17 +25,22 @@ const MessageFieldExtractorActions = React.createClass({
     );
   },
   render() {
-    return (
-      <div className="message-field-actions pull-right">
-        <DropdownButton pullRight
-                     bsSize="xsmall"
-                     title="Select extractor type"
-                     key={1}
-                     id={`select-extractor-type-dropdown-field-${this.props.fieldName}`}>
-          {ExtractorUtils.EXTRACTOR_TYPES.map(extractorType => this._formatExtractorMenuItem(extractorType))}
-        </DropdownButton>
-      </div>
-    );
+    let messageField = this.props.message.fields[this.props.fieldName];
+    if(typeof messageField === 'string') {
+      return (
+          <div className="message-field-actions pull-right">
+            <DropdownButton pullRight
+                            bsSize="xsmall"
+                            title="Select extractor type"
+                            key={1}
+                            id={`select-extractor-type-dropdown-field-${this.props.fieldName}`}>
+                {ExtractorUtils.EXTRACTOR_TYPES.map(extractorType => this._formatExtractorMenuItem(extractorType))}
+            </DropdownButton>
+          </div>
+        );
+    } else {
+      return (<div className="message-field-actions pull-right"/>);
```

Comment:
👍 In this case I think it makes a big difference if you know the system well or not. In our case, we know why the dropdown is not shown, so it may actually save us time, as we know which fields are not strings very easily. On the other hand, for somebody just starting to use Graylog, they would be confused because some fields are missing the dropdown for non-obvious reasons. So far we have (mostly) aligned with the second "persona".